ecs-agent/acs/session/task_manifest_responder_test.go

```diff
@@ -0,0 +1,131 @@
+//go:build unit
+// +build unit
+
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package session
+
+import (
+	"testing"
+
+	"github.com/aws/amazon-ecs-agent/ecs-agent/acs/model/ecsacs"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/metrics"
+	"github.com/aws/amazon-ecs-agent/ecs-agent/wsclient"
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/golang/mock/gomock"
+	"github.com/stretchr/testify/require"
+)
+
+// fakeContainerInstanceDirector imitates the ContainerInstanceDirector.
+type fakeContainerInstanceDirector struct {
+	fakeCI fakeContainerInstance
+}
+
+// fakeContainerInstance imitates the ContainerInstance.
+type fakeContainerInstance struct {
+	LatestTaskManifestSequenceNumber int64
+	ManifestMessageID                string
+}
+
+type fakeTaskComparer struct {
+}
+
+func (ftc fakeTaskComparer) CompareRunningTasksOnInstanceWithManifest(tmr *taskManifestResponder, message *ecsacs.TaskManifestMessage) ([]*ecsacs.TaskIdentifier, error) {
+	var empty []*ecsacs.TaskIdentifier
+	return empty, nil
+}
+
+func (cid *fakeContainerInstanceDirector) GetLatestSequenceNumber() int64 {
+	return cid.fakeCI.LatestTaskManifestSequenceNumber
+}
+
+func (cid *fakeContainerInstanceDirector) GetMessageID() string {
+	return cid.fakeCI.ManifestMessageID
+}
+
+func (cid *fakeContainerInstanceDirector) SetLatestSequenceNumber(message *ecsacs.TaskManifestMessage) error {
+	cid.fakeCI.LatestTaskManifestSequenceNumber = aws.Int64Value(message.Timeline)
+	return nil
+}
+
+func (cid *fakeContainerInstanceDirector) SetMessageID(messageID string) error {
+	cid.fakeCI.ManifestMessageID = messageID
+	return nil
+}
+
+// setupTestTaskManifestResponder initializes the TaskManifestResponder for testing.
+func setupTestTaskManifestResponder(t *testing.T, f wsclient.RespondFunc) *taskManifestResponder {
+	// Create task manifest responder.
+	cluster := "test-cluster"
+	ftc := fakeTaskComparer{}
+	cid := fakeContainerInstanceDirector{}
+	cid.fakeCI.LatestTaskManifestSequenceNumber = 15
+	cid.fakeCI.ManifestMessageID = ""
+	testTaskManifestResponder := NewTaskManifestResponder(ftc, &cid, &cid, cluster, metrics.NewNopEntryFactory(), f)
```

Comment:
How about using a mock implementation of metrics factory? That way we can assert on the interactions with the metrics factory.